src/main.rs

```diff
@@ -12,259 +11,56 @@ extern crate pal_pc as pal;
-    pal::init();
-    let mut trussed_clients = pal::trussed::init(&["fobnail_client"]);
-    let mut trussed_fobnail_client = trussed_clients.pop().unwrap();
-
-    let mut neighbor_cache_storage: [Option<(IpAddress, Neighbor)>; 16] = [None; 16];
-    let neighbor_cache = NeighborCache::new(&mut neighbor_cache_storage[..]);
-
-    let mut socket_set = SocketSet::new(vec![]);
-
-    // Set Fobnail's IP address to 169.254.0.1
-    let mut ip_addrs = [IpCidr::new(IpAddress::v4(169, 254, 0, 1), 16)];
-    let eth_phy = pal::ethernet::create_phy();
-    let mut iface = EthernetInterfaceBuilder::new(eth_phy)
-        .ethernet_addr(pal::ethernet::get_ethernet_address())
-        .neighbor_cache(neighbor_cache)
-        .ip_addrs(&mut ip_addrs[..])
-        .finalize();
-
-    let socket = make_udp_socket(ECHO_SERVER_BUF_LEN, 9400);
-    let mut echo_buf = [0u8; 512];
-    debug!("UDP socket initialized");
-
-    let coap_socket = make_udp_socket(COAP_CLIENT_BUF_LEN, COAP_LOCAL_PORT);
-    debug!("COAP socket initialized");
-
-    // Add sockets into socket set
-    let echo_socket_handle = socket_set.add(socket);
-    let coap_socket_handle = socket_set.add(coap_socket);
-
-    let is_token_provisioned = is_token_provisioned(&mut trussed_fobnail_client);
-    let have_rims = have_rims(&mut trussed_fobnail_client);
-    let mut operation_mode = OperationMode::new(&mut trussed_fobnail_client);
+#[pal::main]
+async fn main() {
+    info!("Hello from main");
 
-    operation_mode = if !is_token_provisioned {
-        operation_mode.token_provisioning()
-    } else if !have_rims {
-        operation_mode.provisioning()
-    } else {
-        operation_mode.attestation()
-    };
+    let stack = pal::net::stack();
+    let mut rx_buffer = [0; 4096];
+    let mut tx_buffer = [0; 4096];
+    let mut buf = [0; 4096];
 
-    let mut last_button_state = false;
-    let mut button_press_time = 0;
     loop {
-        let now = pal::timer::get_time_ms() as u64;
-        // Pressing button for 10 s triggers Fobnail Token provisioning.
-        if now - button_press_time > 10000
-            && !matches!(operation_mode, OperationMode::TokenProvisioning(_))
-            && last_button_state
-        {
-            for _ in 0..3 {
-                let t = pal::timer::get_time_ms() as u64;
-                led::control(Led::Green, true);
-                while pal::timer::get_time_ms() as u64 - t < 100 {}
-                let t = pal::timer::get_time_ms() as u64;
-                led::control(Led::Green, false);
-                while pal::timer::get_time_ms() as u64 - t < 100 {}
-            }
-
-            // Clear persistent storage and reboot. After reboot Fobnail
-            // enters provisioning mode.
-            pal::trussed::reset_device(operation_mode.reclaim_trussed());
+        info!("waiting for connection");
+        let mut socket = TcpSocket::new(stack, &mut rx_buffer, &mut tx_buffer);
+        if let Err(e) = socket.accept(1234).await {
```

Comment:
Please change this port to something else, QEMU uses it for GDB server.

Reply:
Done